common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -439,44 +467,57 @@ LoopDispatch:
 			continue LoopDispatch
 		}
 
-		now := s.timeSource.Now()
-		reservation := rateLimiter.Reserve(
-			now,
-			channel.rateLimitRequest,
-		)
-		if reservation.DelayFrom(now) != 0 {
-			reservation.CancelAt(now)
-			channel.throttled = true
-			numThrottled++
-			if numThrottled == len(iwrrChannels.channels) {
-				// all channels throttled
-				break LoopDispatch
-			}
+		if len(channel.Chan()) == 0 {
 			continue LoopDispatch
```

Comment:
Not for this PR, but this is busy waiting. Why aren't we just reading from the channel?

Reply:
Synced offline. `continue Label` will continue to the next iteration, not starting the entire loop from very beginning.